tests/foreman/performance/common/perf_concurrent_test.py

```diff
@@ -0,0 +1,341 @@
+#!/usr/bin/python
+from robottelo.common import ssh
+
+import unittest
+import logging
+import csv
+
+from tests.foreman.performance.common.perf_thread import SubscribeAKPerfThread
+from tests.foreman.performance.common.perf_thread import SubscribeAttachPerfThread
+from tests.foreman.performance.common.perf_thread import DeletePerfThread
+from tests.foreman.performance.common.perf_stat import generate_stat_for_concurrent_thread
+
+class ConcurrentTestCase(unittest.TestCase):
+    """Parent Class of each concurrent system subscription and deletion test case"""
+
+    @classmethod
+    def setUpClass(cls):
+        # super(ConcurrentTestCase, cls).setUpClass()
+        cls.logger = logging.getLogger('robottelo')
+        cls.logger.setLevel(logging.DEBUG)
+
+        """general running parameters"""
+        cls.num_threads = [1, 2, 4, 6, 8, 10]
+        cls.num_buckets = 10 # number of buckets in stat file;
+
+    def setUp(self):
+        self.logger.debug("Running test %s/%s", type(self).__name__, self._testMethodName)
+
+    """
+    Set file handler
+    """
+    @classmethod
+    def set_file_handler(cls):
+        """Set up logger files"""
+        # create new log file handler and set level to debug
+        fh = logging.FileHandler('{}'.format(cls.log_file_name)) # note: change log file name
+        fh.setLevel(logging.DEBUG)
+        # create formatter
+        formatter = logging.Formatter('%(asctime)s : %(levelname)s : %(message)s')
+        # add formater to fh
+        fh.setFormatter(formatter)
+        # add fh to logger
+        cls.logger.addHandler(fh)
+
+    """    
+    Restore from savepoint
+    """
+    def _restore_from_savepoint(self, savepoint):
+        self.logger.info('---------------------------------')
+        self.logger.info("Reset db from /home/backup/{}".format(savepoint))
+        self.logger.info('---------------------------------')
+        # result = ssh.command('source /root/reset-db.sh /home/backup/{}'.format(savepoint))
+
+
+    """
+    Set # of iterations each thread will conduct
+    @para: total_iterations that a test case would conduct
+    @para: current_num_threads
```

Comment:
We use Sphinx to generate documentation. As a result, the docstring should be valid reStructuredText plus a few (reST-compatible) additional Sphinx directives. The result will look something like this:

```python
"""Set the number of iterations each thread will conduct.

For example::

    valid python code
    goes here

:param total_iterations: An integer. The number of iterations a test case should condut.
:param current_num_threads: I have no idea because this docstring doesn't say.

"""
```